File: taska-backend/src/main/java/com/taska/task/adapter/http/TaskCreateRequest.java
package com.taska.task.adapter.http;

import com.taska.task.model.TaskType;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

/** Request payload containing only properties accepted when creating a task. */
public record TaskCreateRequest(@NotBlank String content, String description, UUID projectId, UUID parentId, Integer order,
        @Min(1) @Max(4) Integer priority, List<String> labels, Instant scheduledAt, Instant dueAt, Boolean allDay, Boolean isRecurring,
        @Positive Integer estimateMinutes, String mentionContext, String recurrenceRule, TaskType type) {
}

File: taska-android/app/src/main/java/com/taska/android/data/model/TaskCreateRequest.kt
package com.taska.android.data.model

data class TaskCreateRequest(
    val content: String,
    val type: String? = null,
    val description: String? = null,
    val projectId: String? = null,
    val parentId: String? = null,
    val order: Int? = null,
    val priority: Int? = null,
    val labels: List<String>? = null,
    val scheduledAt: String? = null,
    val dueAt: String? = null,
    val allDay: Boolean? = null,
    val estimateMinutes: Int? = null,
    val mentionContext: String? = null,
    val isRecurring: Boolean? = null,
    val recurrenceRule: String? = null,
)
